Src/DependencyCollector/Shared/Implementation/ProfilerHttpProcessing.cs

```diff
@@ -385,13 +391,17 @@ private void OnEnd(object exception, object thisObj, object returnValue)
 
                             if (responseObj.Headers != null)
                             {
-                                var targetIkeyHash = responseObj.Headers[RequestResponseHeaders.TargetInstrumentationKeyHeader];
-                                
-                                // We only add the cross component correlation key if the key does not remain the current component.
-                                if (!string.IsNullOrEmpty(targetIkeyHash) && targetIkeyHash != InstrumentationKeyHashLookupHelper.GetInstrumentationKeyHash(telemetry.Context.InstrumentationKey))
+                                var targetAppId = responseObj.Headers[RequestResponseHeaders.TargetAppIdHeader];
+
+                                string myAppId;
+                                if (CorrelationIdLookupHelper.TryGetXComponentCorrelationId(telemetry.Context.InstrumentationKey, this.appIdEndpoint, out myAppId))
                                 {
-                                    telemetry.Type = RemoteDependencyConstants.AI;
-                                    telemetry.Target += " | " + targetIkeyHash;
+                                    // We only add the cross component correlation key if the key does not remain the current component.
+                                    if (!string.IsNullOrEmpty(targetAppId) && targetAppId != myAppId)
+                                    {
+                                        telemetry.Type = RemoteDependencyConstants.AI;
+                                        telemetry.Target += " | " + targetAppId;
+                                    }
                                 }
                             }
                         }
```

Comment:
The try method shouldn't throw. I've updated the implementation. With that, we shouldn't need to change anything.